src/Microsoft.VisualStudio.Services.Agent/ConfigurationStore.cs

```diff
@@ -43,7 +43,7 @@ public sealed class AgentSettings
         public string ProjectName { get; set; }
 
         [DataMember(EmitDefaultValue = false)]
-        public string MachineGroupName { get; set; }
+        public int MachineGroupId { get; set; }
```

Comment:
do you need to worry about people who already configure the agent that has machinegroupname in the .agent?